snippets/Google.Storage.V1.Snippets/StorageClientSnippets.cs

```diff
@@ -204,6 +266,175 @@ public void GetObject()
+            Assert.Equal("value1", fetchedObject.Metadata["key1"]);
+            Assert.Equal("updated-value2", fetchedObject.Metadata["key2"]);
+            Assert.Equal("value3", fetchedObject.Metadata["key3"]);
+        }
+
+        // TODO: Do we want to expose CopyObject, or just always use RewriteObject?
+
+        [Fact]
+        public void CopyObject()
+        {
+            var projectId = _fixture.ProjectId;
+            var sourceBucket = _fixture.BucketName;
+            var destinationBucket = Guid.NewGuid().ToString();
+            StorageClient.Create().CreateBucket(projectId, destinationBucket);
+            _fixture.RegisterBucketToDelete(destinationBucket);
+
+            // <CopyObject>
+            var client = StorageClient.Create();
+            var sourceName = "greetings/hello.txt";
+            var destinationName = "copy.txt";
+
+            client.Service.Objects.Copy(null, sourceBucket, sourceName, destinationBucket, destinationName).Execute();
+            // </CopyObject>
+
+            var obj = client.GetObject(destinationBucket, destinationName);
+            Assert.Equal((ulong) Encoding.UTF8.GetByteCount(_fixture.HelloWorldContent), obj.Size.Value);
+        }
+
+        [Fact]
+        public void RewriteObject()
+        {
+            var projectId = _fixture.ProjectId;
+            var sourceBucket = _fixture.BucketName;
+            var destinationBucket = Guid.NewGuid().ToString();
+            StorageClient.Create().CreateBucket(projectId, destinationBucket);
+            _fixture.RegisterBucketToDelete(destinationBucket);
+
+            // <RewriteObject>
+            var client = StorageClient.Create();
+            var sourceName = "greetings/hello.txt";
+            var destinationName = "copy.txt";
+
+            var request = client.Service.Objects.Rewrite(null, sourceBucket, sourceName, destinationBucket, destinationName);
+            var response = request.Execute();
+            while (response.RewriteToken != null)
+            {
+                request.RewriteToken = response.RewriteToken;
+                response = request.Execute();
+            }
+            // </RewriteObject>
+
+            var obj = client.GetObject(destinationBucket, destinationName);
+            Assert.Equal((ulong)Encoding.UTF8.GetByteCount(_fixture.HelloWorldContent), obj.Size.Value);
+        }
+
+        // FIXME: This fails at the moment as we'd need to register a domain for notifications.
+        [Fact]
+        public void WatchBucket()
+        {
+            var projectId = _fixture.ProjectId;
+            var bucket = Guid.NewGuid().ToString();
+            StorageClient.Create().CreateBucket(projectId, bucket);
+            _fixture.RegisterBucketToDelete(bucket);
+
+            // <WatchBucket>
+            var client = StorageClient.Create();
+
+            var channel = client.Service.Objects.WatchAll(new Channel { Address = "https://example.com/notify" }, bucket).Execute();
```

Comment:
You watch them by providing a URL that GCS notifies of changes. The `bucket` part here is specifying the name of the bucket to watch. (You can also specify a prefix, to watch just part of a bucket.)